plugins/catalog-react/src/components/EntityOwnerPicker/EntityOwnerPicker.tsx

```diff
@@ -114,7 +83,13 @@ export const EntityOwnerPicker = () => {
     queryParamOwners.length ? queryParamOwners : filters.owners?.values ?? [],
   );
 
-  const { getEntity, setEntity } = useSelectedOwners(selectedOwners);
+  const [{ value, loading }, handleFetch, cache] = useFetchEntities({
+    mode,
+    initialSelectedOwnersRefs: selectedOwners,
+  });
+  useDebouncedEffect(() => handleFetch({ text }), [text, handleFetch], 250);
```

Comment:
I believe we want to tweak this or something else so that it's more lazy and only loads the set of owners when you actually open the dropdown right? I'm assuming that the eagerness of this is also what's breaking the e2e tests?